### ISSUE TITLE
Pylint does not warn about missing docstrings for inner functions

### ISSUE THREAD

**Issue by mustafa-cosar** [PyCQA/pylint#3177]
Hi, I searched for this issue but couldn't find it.

The problem is with `missing-function-docstring` rule. 

### Steps to reproduce
1. Consider the following code in a Python file
```py
def foo():
    def bar():
        pass
```
2. Running pylint produces the following output:
```
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
test.py:1:0: C0102: Black listed name "foo" (blacklisted-name)
test.py:1:0: C0116: Missing function or method docstring (missing-function-docstring)
test.py:2:4: C0102: Black listed name "bar" (blacklisted-name)
test.py:2:4: W0612: Unused variable 'bar' (unused-variable)

--------------------------------------------------------------------
Your code has been rated at -6.67/10 (previous run: -6.67/10, +0.00)
```

### Current behavior
Pylint does not give an error for `bar()` inner function

### Expected behavior
Pylint gives `missing-function-docstring` error for `bar()` inner function, similar to `foo()`

### pylint --version output
pylint 2.4.1
astroid 2.3.0
Python 3.7.4 (default, Sep 13 2019, 14:42:01) 
[GCC 7.4.0]

**Comment by PCManticore** [comment]
Thanks for opening an issue. This message is emitted only for public APIs such as top level functions, classes and modules. It's not emitted for private objects (e.g. with a prepended underscore) nor for internal objects like this one.

**Comment by mustafa-cosar** [comment]
Thanks for the quick response! 

I understand your point, however this wasn't always the case. We were using pylint for a long time and recently updated from `1.9.5` to `2.4.1`. Are you sure this isn't a regression? I looked at the changelogs and saw that the `missing-docstring` was separated to module/class/function docstrings so it may be related to that.

Is it possible to request this as a feature? We would like no function to go without a docstring.

Thanks again

**Comment by mustafa-cosar** [comment]
Ping @PCManticore

**Comment by PCManticore** [comment]
@mustafa-cosar Maybe this wasn't always the case but the current behaviour makes more sense for what the check is trying to do. Extending it to encompass all functions, even closures, is not something I'd like to change.
